From b5fc95868141b5526947e6037103e21b81f31b92 Mon Sep 17 00:00:00 2001
From: Tobias Klinke <TobiasAlexander.Klinke@baslerweb.com>
Date: Fri, 12 Feb 2021 12:08:41 +0100
Subject: [PATCH 20/24] media: mxc: change mx6s_fmt.bpp to bits per pixel

The previous meaning was bytes per pixel.
This commit prepares for adding YUV420 support
which will be 12 bits per pixel.

Applied from https://github.com/basler/meta-basler-imx8/tree/b1b6d6653931303bffbad727d767fe5a8d4844fc/recipes-kernel/linux/linux-imx
Adopted to local changes.

Signed-off-by: Alexander Stein <alexander.stein@ew.tq-group.com>
---
 .../media/platform/mxc/capture/mx6s_capture.c | 40 +++++++++----------
 1 file changed, 20 insertions(+), 20 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/mx6s_capture.c b/drivers/media/platform/mxc/capture/mx6s_capture.c
index f26b1515ebf5..d506cca4712b 100644
--- a/drivers/media/platform/mxc/capture/mx6s_capture.c
+++ b/drivers/media/platform/mxc/capture/mx6s_capture.c
@@ -276,109 +276,109 @@ static struct mx6s_fmt formats[] = {
 		.fourcc		= V4L2_PIX_FMT_UYVY,
 		.pixelformat	= V4L2_PIX_FMT_UYVY,
 		.mbus_code	= MEDIA_BUS_FMT_UYVY8_2X8,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "YUYV-16",
 		.fourcc		= V4L2_PIX_FMT_YUYV,
 		.pixelformat	= V4L2_PIX_FMT_YUYV,
 		.mbus_code	= MEDIA_BUS_FMT_YUYV8_2X8,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "YUV32 (X-Y-U-V)",
 		.fourcc		= V4L2_PIX_FMT_YUV32,
 		.pixelformat	= V4L2_PIX_FMT_YUV32,
 		.mbus_code	= MEDIA_BUS_FMT_AYUV8_1X32,
-		.bpp		= 4,
+		.bpp		= 32,
 	}, {
 		.name		= "SBGGR8",
 		.fourcc		= V4L2_PIX_FMT_SBGGR8,
 		.pixelformat	= V4L2_PIX_FMT_SBGGR8,
 		.mbus_code	= MEDIA_BUS_FMT_SBGGR8_1X8,
-		.bpp		= 1,
+		.bpp		= 8,
 	}, {
 		.name		= "SBGGR10",
 		.fourcc		= V4L2_PIX_FMT_SBGGR10,
 		.pixelformat	= V4L2_PIX_FMT_SBGGR10,
 		.mbus_code	= MEDIA_BUS_FMT_SBGGR10_1X10,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "SBGGR12",
 		.fourcc		= V4L2_PIX_FMT_SBGGR12,
 		.pixelformat	= V4L2_PIX_FMT_SBGGR12,
 		.mbus_code	= MEDIA_BUS_FMT_SBGGR12_1X12,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "SGRBG8",
 		.fourcc		= V4L2_PIX_FMT_SGRBG8,
 		.pixelformat	= V4L2_PIX_FMT_SGRBG8,
 		.mbus_code	= MEDIA_BUS_FMT_SGRBG8_1X8,
-		.bpp		= 1,
+		.bpp		= 8,
 	}, {
 		.name		= "SGRBG10",
 		.fourcc		= V4L2_PIX_FMT_SGRBG10,
 		.pixelformat	= V4L2_PIX_FMT_SGRBG10,
 		.mbus_code	= MEDIA_BUS_FMT_SGRBG10_1X10,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "SGRBG12",
 		.fourcc		= V4L2_PIX_FMT_SGRBG12,
 		.pixelformat	= V4L2_PIX_FMT_SGRBG12,
 		.mbus_code	= MEDIA_BUS_FMT_SGRBG12_1X12,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "SGBRG8",
 		.fourcc		= V4L2_PIX_FMT_SGBRG8,
 		.pixelformat	= V4L2_PIX_FMT_SGBRG8,
 		.mbus_code	= MEDIA_BUS_FMT_SGBRG8_1X8,
-		.bpp		= 1,
+		.bpp		= 8,
 	}, {
 		.name		= "SGBRG10",
 		.fourcc		= V4L2_PIX_FMT_SGBRG10,
 		.pixelformat	= V4L2_PIX_FMT_SGBRG10,
 		.mbus_code	= MEDIA_BUS_FMT_SGBRG10_1X10,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "SGBRG12",
 		.fourcc		= V4L2_PIX_FMT_SGBRG12,
 		.pixelformat	= V4L2_PIX_FMT_SGBRG12,
 		.mbus_code	= MEDIA_BUS_FMT_SGBRG12_1X12,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "SRGGB8",
 		.fourcc		= V4L2_PIX_FMT_SRGGB8,
 		.pixelformat	= V4L2_PIX_FMT_SRGGB8,
 		.mbus_code	= MEDIA_BUS_FMT_SRGGB8_1X8,
-		.bpp		= 1,
+		.bpp		= 8,
 	}, {
 		.name		= "SRGGB10",
 		.fourcc		= V4L2_PIX_FMT_SRGGB10,
 		.pixelformat	= V4L2_PIX_FMT_SRGGB10,
 		.mbus_code	= MEDIA_BUS_FMT_SRGGB10_1X10,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "SRGGB12",
 		.fourcc		= V4L2_PIX_FMT_SRGGB12,
 		.pixelformat	= V4L2_PIX_FMT_SRGGB12,
 		.mbus_code	= MEDIA_BUS_FMT_SRGGB12_1X12,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "Grey8 (Y8)",
 		.fourcc		= V4L2_PIX_FMT_GREY,
 		.pixelformat	= V4L2_PIX_FMT_GREY,
 		.mbus_code	= MEDIA_BUS_FMT_Y8_1X8,
-		.bpp		= 1,
+		.bpp		= 8,
 	}, {
 		.name		= "Grey10 (Y10)",
 		.fourcc		= V4L2_PIX_FMT_Y10,
 		.pixelformat	= V4L2_PIX_FMT_Y10,
 		.mbus_code	= MEDIA_BUS_FMT_Y10_1X10,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "Grey12 (Y12)",
 		.fourcc		= V4L2_PIX_FMT_Y12,
 		.pixelformat	= V4L2_PIX_FMT_Y12,
 		.mbus_code	= MEDIA_BUS_FMT_Y12_1X12,
-		.bpp		= 2,
+		.bpp		= 16,
 	}
 };
 
@@ -1660,8 +1660,8 @@ static int _mx6s_vidioc_try_fmt_vid_cap(struct file *file, void *priv,
 	if (pix->field != V4L2_FIELD_INTERLACED)
 		pix->field = V4L2_FIELD_NONE;
 
-	pix->sizeimage = fmt->bpp * pix->height * pix->width;
-	pix->bytesperline = fmt->bpp * pix->width;
+	pix->bytesperline = DIV_ROUND_UP(fmt->bpp * pix->width, 8);
+	pix->sizeimage = pix->bytesperline * pix->height;
 
 	pix->colorspace = V4L2_COLORSPACE_SRGB;
 	pix->ycbcr_enc = V4L2_MAP_YCBCR_ENC_DEFAULT(pix->colorspace);
-- 
2.25.1

